### ISSUE TITLE
💅 Incorrectly flagging noUselessFragments

### ISSUE THREAD

**Issue by camerongineer** [biomejs/biome#8002]
### Environment information

```bash
CLI:
  Version:                      2.3.4
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           v22.21.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/10.9.4

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0
```

### Rule name

lint/complexity/noUselessFragments

### Playground link

https://biomejs.dev/playground/?code=IgB1AHMAZQAgAG4AbwAgAG0AZQBtAG8AIgAKAGkAbQBwAG8AcgB0ACAAewAgAFQAeQBwAG8AZwByAGEAcABoAHkAIAB9ACAAZgByAG8AbQAgACIAQABtAHUAaQAvAG0AYQB0AGUAcgBpAGEAbAAiADsACgBpAG0AcABvAHIAdAAgAHsAIAB1AHMAZQBQAHIAbwBnAHIAYQBtAFQAeQBwAGUAcwBRAHUAZQByAHkAIAB9ACAAZgByAG8AbQAgACIALgAuAC8ALgAuAC8ALgAuAC8ASABvAG8AawBzAC8AdQBzAGUAUAByAG8AZwByAGEAbQBUAHkAcABlAHMAUQB1AGUAcgB5ACIAOwAKAAoAYwBvAG4AcwB0ACAAUAByAG8AZwByAGEAbQBUAHkAcABlAFMAZQBsAGUAYwB0AG8AcgAgAD0AIAAoACkAIAA9AD4AIAB7AAoAIAAgACAAIABjAG8AbgBzAHQAIAB7ACAAcAByAG8AZwByAGEAbQBUAHkAcABlAHMAIAB9ACAAPQAgAHUAcwBlAFAAcgBvAGcAcgBhAG0AVAB5AHAAZQBzAFEAdQBlAHIAeQAoACkAOwAKAAoAIAAgACAAIABjAG8AbgBzAHQAIAB0AHkAcABlAHMAVwBpAHQAaABOAG8AdABlAHMAIAA9ACAAcAByAG8AZwByAGEAbQBUAHkAcABlAHMALgBmAGkAbAB0AGUAcgAoACgAcAB0ACkAIAA9AD4AIABwAHQALgBuAG8AdABlACkAOwAKACAAIAAgACAAYwBvAG4AcwB0ACAAbgBvAHQAZQBNAGEAcAAgAD0AIABuAGUAdwAgAE0AYQBwADwAbgB1AG0AYgBlAHIALAAgAG4AdQBtAGIAZQByAD4AKAApADsACgAgACAAIAAgAHQAeQBwAGUAcwBXAGkAdABoAE4AbwB0AGUAcwAuAGYAbwByAEUAYQBjAGgAKAAoAHAAdAAsACAAaQBuAGQAZQB4ACkAIAA9AD4AIAB7AAoAIAAgACAAIAAgACAAIAAgAG4AbwB0AGUATQBhAHAALgBzAGUAdAAoAHAAdAAuAHAAcgBvAGcAcgBhAG0AVAB5AHAAZQBJAGQALAAgAGkAbgBkAGUAeAAgACsAIAAxACkAOwAKACAAIAAgACAAfQApADsACgAKACAAIAAgACAAcgBlAHQAdQByAG4AIAAoAAoAIAAgACAAIAAgACAAIAAgADwAPgAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgAHsAdAB5AHAAZQBzAFcAaQB0AGgATgBvAHQAZQBzAC4AbABlAG4AZwB0AGgAIAA%2BACAAMAAgACYAJgAgACgACgAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgADwAPgAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAB7AHQAeQBwAGUAcwBXAGkAdABoAE4AbwB0AGUAcwAuAG0AYQBwACgAKABwAHIAbwBnAHIAYQBtAFQAeQBwAGUAKQAgAD0APgAgAHsACgAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAByAGUAdAB1AHIAbgAgACgACgAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgADwAVAB5AHAAbwBnAHIAYQBwAGgAeQAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIABrAGUAeQA9AHsAcAByAG8AZwByAGEAbQBUAHkAcABlAC4AcAByAG8AZwByAGEAbQBUAHkAcABlAEkAZAB9AAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAA%2BAAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgAHsAcAByAG8AZwByAGEAbQBUAHkAcABlAC4AbgBvAHQAZQB9AAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAA8AC8AVAB5AHAAbwBnAHIAYQBwAGgAeQA%2BAAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAKQA7AAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgAH0AKQB9AAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAA8AC8APgAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgACkAfQAKACAAIAAgACAAIAAgACAAIAA8AC8APgAKACAAIAAgACAAKQA7AAoAfQA7AAoACgBlAHgAcABvAHIAdAAgAGQAZQBmAGEAdQBsAHQAIABQAHIAbwBnAHIAYQBtAFQAeQBwAGUAUwBlAGwAZQBjAHQAbwByADsACgA%3D

### Expected result

There should not be a warning because the fragment is necessary

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by siketyan** [comment]
Can you explain why the fragment is necessary here? I think it's unnecessary in this case.

**Comment by camerongineer** [comment]
@siketyan Remove the fragment in the playground and you see the types of errors you will also is in an IDE. In this case if you wish to use parenthesis to encapsulate the mapping, React does not allow this without the fragment. So technically it's not useless. But I do know there are other ways around doing it this way, though. Just seems like improper flagging to me.

**Comment by siketyan** [comment]
@camerongineer Isn't this a valid code after removing the fragment?

https://biomejs.dev/playground/?code=IgB1AHMAZQAgAG4AbwAgAG0AZQBtAG8AIgAKAGkAbQBwAG8AcgB0ACAAewAgAFQAeQBwAG8AZwByAGEAcABoAHkAIAB9ACAAZgByAG8AbQAgACIAQABtAHUAaQAvAG0AYQB0AGUAcgBpAGEAbAAiADsACgBpAG0AcABvAHIAdAAgAHsAIAB1AHMAZQBQAHIAbwBnAHIAYQBtAFQAeQBwAGUAcwBRAHUAZQByAHkAIAB9ACAAZgByAG8AbQAgACIALgAuAC8ALgAuAC8ALgAuAC8ASABvAG8AawBzAC8AdQBzAGUAUAByAG8AZwByAGEAbQBUAHkAcABlAHMAUQB1AGUAcgB5ACIAOwAKAAoAYwBvAG4AcwB0ACAAUAByAG8AZwByAGEAbQBUAHkAcABlAFMAZQBsAGUAYwB0AG8AcgAgAD0AIAAoACkAIAA9AD4AIAB7AAoAIAAgACAAIABjAG8AbgBzAHQAIAB7ACAAcAByAG8AZwByAGEAbQBUAHkAcABlAHMAIAB9ACAAPQAgAHUAcwBlAFAAcgBvAGcAcgBhAG0AVAB5AHAAZQBzAFEAdQBlAHIAeQAoACkAOwAKAAoAIAAgACAAIABjAG8AbgBzAHQAIAB0AHkAcABlAHMAVwBpAHQAaABOAG8AdABlAHMAIAA9ACAAcAByAG8AZwByAGEAbQBUAHkAcABlAHMALgBmAGkAbAB0AGUAcgAoACgAcAB0ACkAIAA9AD4AIABwAHQALgBuAG8AdABlACkAOwAKACAAIAAgACAAYwBvAG4AcwB0ACAAbgBvAHQAZQBNAGEAcAAgAD0AIABuAGUAdwAgAE0AYQBwADwAbgB1AG0AYgBlAHIALAAgAG4AdQBtAGIAZQByAD4AKAApADsACgAgACAAIAAgAHQAeQBwAGUAcwBXAGkAdABoAE4AbwB0AGUAcwAuAGYAbwByAEUAYQBjAGgAKAAoAHAAdAAsACAAaQBuAGQAZQB4ACkAIAA9AD4AIAB7AAoAIAAgACAAIAAgACAAIAAgAG4AbwB0AGUATQBhAHAALgBzAGUAdAAoAHAAdAAuAHAAcgBvAGcAcgBhAG0AVAB5AHAAZQBJAGQALAAgAGkAbgBkAGUAeAAgACsAIAAxACkAOwAKACAAIAAgACAAfQApADsACgAKACAAIAAgACAAcgBlAHQAdQByAG4AIAAoAAoAIAAgACAAIAAgACAAIAAgADwAPgAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgAHsAdAB5AHAAZQBzAFcAaQB0AGgATgBvAHQAZQBzAC4AbABlAG4AZwB0AGgAIAA%2BACAAMAAgACYAJgAgACgACgAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgAHQAeQBwAGUAcwBXAGkAdABoAE4AbwB0AGUAcwAuAG0AYQBwACgAKABwAHIAbwBnAHIAYQBtAFQAeQBwAGUAKQAgAD0APgAgAHsACgAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAByAGUAdAB1AHIAbgAgACgACgAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgADwAVAB5AHAAbwBnAHIAYQBwAGgAeQAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIABrAGUAeQA9AHsAcAByAG8AZwByAGEAbQBUAHkAcABlAC4AcAByAG8AZwByAGEAbQBUAHkAcABlAEkAZAB9AAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAA%2BAAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgAHsAcAByAG8AZwByAGEAbQBUAHkAcABlAC4AbgBvAHQAZQB9AAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAA8AC8AVAB5AHAAbwBnAHIAYQBwAGgAeQA%2BAAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAKQA7AAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgAH0AKQAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgACkAfQAKACAAIAAgACAAIAAgACAAIAA8AC8APgAKACAAIAAgACAAKQA7AAoAfQA7AAoACgBlAHgAcABvAHIAdAAgAGQAZQBmAGEAdQBsAHQAIABQAHIAbwBnAHIAYQBtAFQAeQBwAGUAUwBlAGwAZQBjAHQAbwByADsACgA%3D

**Comment by camerongineer** [comment]
@siketyan Yes, fair enough. And I also know you can do it without parenthesis as well. I guess my issue was with it flagging it as useless when technically had to be there unless remove other code as well (in this case the curly braces) otherwise it breaks the code. More of a stylist issue on my part I suppose.
